src/main/java/uk/gov/hmcts/ccd/data/casedetails/DefaultCaseDetailsRepository.java

```diff
@@ -157,6 +174,27 @@ public PaginatedSearchMetadata getPaginatedSearchMetadata(MetaData metaData, Map
         return sr;
     }
 
+    // TODO This accepts null values for backward compatibility. Once deprecated methods are removed, parameters should
+    // be annotated with @NotNull
+    private Optional<CaseDetailsEntity> find(String jurisdiction, Long id, String reference) {
+        final CaseDetailsQueryBuilder qb = queryBuilderFactory.select(em);
+
+        if (null != jurisdiction) {
+            qb.whereJurisdiction(jurisdiction);
+        }
+
+        if (null != reference) {
+            qb.whereReference(reference);
+        } else {
+            qb.whereId(id);
+        }
+
+        final TypedQuery<CaseDetailsEntity> query = qb.build();
```

Comment:
can we remove final please

Reply:
Removed `query` variable and used `qb.build()` instead.